Model/Api/DiscountCodeValidation.php

```diff
@@ -800,7 +790,7 @@ private function getCartTotals($quote)
     {
         $request = $this->getRequestContent();
         $is_has_shipment = isset($request->cart->shipments[0]->reference);
```

Comment:
Note: shipments is populated for the user logged in Bolt checkout (shipping and tax is called every time in the process of generating checkout data) and also for the guest users after they apply the discount in the modal (shipping and tax gets called by Bolt as a side effect of discount code validation). No matter if the cart is virtual or not. However shipping and tax endpoint is not called for guest user virtual carts in general (no coupon applied). This is inconsistency in Bolt functionality that causes issues and should be fixed on Bolt end.